core/src/com/mygdx/game/client/screen/ChooseUnitFieldDialog.java

```diff
@@ -0,0 +1,58 @@
+package com.mygdx.game.client.screen;
+
+import com.badlogic.ashley.core.ComponentMapper;
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Dialog;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.mygdx.game.client.component.FieldComponent;
+import com.mygdx.game.client.component.UnitComponent;
+import lombok.extern.java.Log;
+
+import java.util.logging.Level;
+
+@Log
+public class ChooseUnitFieldDialog {
+
+  private static final ComponentMapper<FieldComponent> fieldComponentMapper = ComponentMapper.getFor(FieldComponent.class);
+  private static final ComponentMapper<UnitComponent> unitComponentMapper = ComponentMapper.getFor(UnitComponent.class);
+  private static final Skin uiSkin = new Skin(Gdx.files.internal("dialog/custom/uiskin.json"));
+
+  public static Dialog customDialog = null;
+
+  public static void createCustomDialog(Stage parentStage, Entity entity) {
+    if (customDialog != null) {
+      customDialog.hide();
+      customDialog = null;
+    }
+
+
+    customDialog = new Dialog("Choose", uiSkin) {
+      @Override
+      protected void result(Object object) {
+        super.result(object);
+        if (object.equals("field")) {
+          log.log(Level.INFO, "Selected field.");
+        } else if (object.equals("unit")) {
```

Comment:
może warto do createCustomDialog przekazywać callback, który byłby wywoływany w result? może wtedy będzie się działo użyć generyków żeby naprawić trochę api, ale trzeba będzie wyłączyć ostrzeżenie

Reply:
Hmmm narazie tego nie widzę, mógłbyś jakoś to zaproponować?